sample-plugin/pom.xml

```diff
@@ -129,11 +129,21 @@
             <classifier>tests</classifier>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>mailer</artifactId>
```

Comment:
Unnecessary, since it is being pulled in transitively from something else (like `workflow-basic-steps`). If it _were_ necessary, the build would fail with

```
Managed plugin dependency mailer not listed in test classpath of sample plugin
```

Reply:
Is keeping `sample-plugin/pom.xml` as minimal as possible a goal in and of itself? It would turn a simple procedure (add each new plugin to both `bom/pom.xml` and `sample-plugin/pom.xml`) to a more complicated and error-prone one (add each new plugin to `bom/pom.xml`, then check to see if it is transitively consumed by another plugin already or else add it to `sample-plugin/pom.xml`). I think keeping a one-to-one correspondence between `bom/pom.xml` and `sample-plugin/pom.xml` is actually simpler; what do you think?

Reply:
> Is keeping `sample-plugin/pom.xml` as minimal as possible a goal in and of itself?

A minor goal, yes. The `sample-plugin` is supposed to look as much as possible like a realistic plugin would, and developers would not normally be declaring dependencies on things like `git-server` unless they were actually consciously using that plugin. (There is even a goal `mvn dependency:analyze` which is supposed to help automate this, though there are so many special cases in Jenkins with reflective usages that it has to treated solely as a hint.)

> a more complicated and error-prone [procedure]

I do not think so. If you had a reason to add a plugin to the managed set for some reason, typically it was because you found it in a real plugin’s POM. So, add it to `sample-plugin/pom.xml` too, as well as `bom/pom.xml`. If you then get errors saying that some dependencies are unmanaged, add them just to `bom/pom.xml`. If there is ever a situation where a plugin is _missing_ from `sample-plugin/pom.xml`, your build will fail quickly with an error saying so.

Reply:
That makes sense to me. Thanks for the explanation!

Reply:
Maybe needs to be called out more explicitly in `README.md`?

Reply:
Sure, I updated the `README` to include a summary of these points.